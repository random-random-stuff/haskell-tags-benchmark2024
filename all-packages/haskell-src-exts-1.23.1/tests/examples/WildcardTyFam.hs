{-# LANGUAGE TypeFamilies #-}

type family Foo a where
  Foo _ = Int
