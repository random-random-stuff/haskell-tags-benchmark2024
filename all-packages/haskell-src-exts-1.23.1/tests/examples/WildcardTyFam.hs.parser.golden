ParseOk
  ( Module
      SrcSpanInfo
        { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 1 1 5 1
        , srcInfoPoints =
            [ SrcSpan "tests/examples/WildcardTyFam.hs" 1 1 1 1
            , SrcSpan "tests/examples/WildcardTyFam.hs" 3 1 3 1
            , SrcSpan "tests/examples/WildcardTyFam.hs" 3 1 3 1
            , SrcSpan "tests/examples/WildcardTyFam.hs" 3 1 3 1
            , SrcSpan "tests/examples/WildcardTyFam.hs" 5 1 5 1
            , SrcSpan "tests/examples/WildcardTyFam.hs" 5 1 5 1
            ]
        }
      Nothing
      [ LanguagePragma
          SrcSpanInfo
            { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 1 1 1 30
            , srcInfoPoints =
                [ SrcSpan "tests/examples/WildcardTyFam.hs" 1 1 1 13
                , SrcSpan "tests/examples/WildcardTyFam.hs" 1 27 1 30
                ]
            }
          [ Ident
              SrcSpanInfo
                { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 1 14 1 26
                , srcInfoPoints = []
                }
              "TypeFamilies"
          ]
      ]
      []
      [ ClosedTypeFamDecl
          SrcSpanInfo
            { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 3 1 3 18
            , srcInfoPoints =
                [ SrcSpan "tests/examples/WildcardTyFam.hs" 3 1 3 5
                , SrcSpan "tests/examples/WildcardTyFam.hs" 3 6 3 12
                , SrcSpan "tests/examples/WildcardTyFam.hs" 3 19 3 24
                ]
            }
          (DHApp
             SrcSpanInfo
               { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 3 13 3 18
               , srcInfoPoints = []
               }
             (DHead
                SrcSpanInfo
                  { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 3 13 3 16
                  , srcInfoPoints = []
                  }
                (Ident
                   SrcSpanInfo
                     { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 3 13 3 16
                     , srcInfoPoints = []
                     }
                   "Foo"))
             (UnkindedVar
                SrcSpanInfo
                  { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 3 17 3 18
                  , srcInfoPoints = []
                  }
                (Ident
                   SrcSpanInfo
                     { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 3 17 3 18
                     , srcInfoPoints = []
                     }
                   "a")))
          Nothing
          Nothing
          [ TypeEqn
              SrcSpanInfo
                { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 4 3 4 14
                , srcInfoPoints =
                    [ SrcSpan "tests/examples/WildcardTyFam.hs" 4 9 4 10 ]
                }
              (TyApp
                 SrcSpanInfo
                   { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 4 3 4 8
                   , srcInfoPoints = []
                   }
                 (TyCon
                    SrcSpanInfo
                      { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 4 3 4 6
                      , srcInfoPoints = []
                      }
                    (UnQual
                       SrcSpanInfo
                         { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 4 3 4 6
                         , srcInfoPoints = []
                         }
                       (Ident
                          SrcSpanInfo
                            { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 4 3 4 6
                            , srcInfoPoints = []
                            }
                          "Foo")))
                 (TyWildCard
                    SrcSpanInfo
                      { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 4 7 4 8
                      , srcInfoPoints = []
                      }
                    Nothing))
              (TyCon
                 SrcSpanInfo
                   { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 4 11 4 14
                   , srcInfoPoints = []
                   }
                 (UnQual
                    SrcSpanInfo
                      { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 4 11 4 14
                      , srcInfoPoints = []
                      }
                    (Ident
                       SrcSpanInfo
                         { srcInfoSpan = SrcSpan "tests/examples/WildcardTyFam.hs" 4 11 4 14
                         , srcInfoPoints = []
                         }
                       "Int")))
          ]
      ]
  , []
  )
